pkg/store/consul/transaction/transaction.go

```diff
@@ -35,10 +35,20 @@ type tx struct {
 	committedMu sync.Mutex
 }
 
+// New() returns a new context derived from the one passed as an argument and
+// its cancel function with the context containing metadata used to build a
+// consul transaction. If the passed context already has consul operations
+// defined, the new context will inherit those operations defined but in a
+// separate transaction.
 func New(ctx context.Context) (context.Context, context.CancelFunc) {
```

Comment:
Without seeing this used in _context_, it's hard for me to say whether this behaviour overloading is clear or confusing. Could we accomplish the same goal with a `Branch()` constructor?